docs/data-structures.md

```diff
@@ -0,0 +1,249 @@
+    "parent_hash": "0x255f65bf9dc00bcd9f9b8be8624be222cba16b51366208a8267f1925eb40e7e4",
+    "seal": {
+        "nonce": 503529102265201399,
+        "proof": "0x"
+    },
+    "timestamp": 1551155125985,
+    "txs_commit": "0xabeb06aea75b59ec316db9d21243ee3f0b0ad0723e50f57761cef7e07974b9b5",
+    "txs_proposal": "0x0000000000000000000000000000000000000000000000000000000000000000",
+    "uncles_count": 1,
+    "uncles_hash": "0x99cf8710e59303bfac236b57256fcea2c58192f2c9c39d1ea4c19cbcf88b4952",
+    "version": 0
+},
+"proposal_transactions": [],
+"uncles": [
+    {
+    "cellbase": {
+        ...
+    },
+    "header": {
+        ...
+    },
+    "proposal_transactions": []
+    }
+]
+}
+```
+
+### Description
+
+#### Block
+
+| Name                    | Type            | Description                                                  |
+| ----------------------- | --------------- | ------------------------------------------------------------ |
+| `header`                | `Header`        | **The block header of the block.** This part contains some metadata of the block. |
+| `commit_trasactions`    | [`Transaction`] | **An array of transactions contained in the block.** This is where the miner put the received transactions. |
+| `proposal_transactions` | [string]        | **An array of hex-encoded short transaction ID.**            |
+| `uncles`                | [`UncleBlock`]  | **An array of uncle blocks of the block.**                   |
+
+#### Header
+
+| Name           | Type                | Description                                                  |
+| -------------- | ------------------- | ------------------------------------------------------------ |
+| `cellbase_id`  | H256(hash)          | **The hash of the Cellbase transaction.** Cellbase transaction is just like the coinbase transaction in Bitcoin. It's the transaction added by the miner who mined this block, by which the miner receives block reward for successfully mined the block. |
+| `difficulty`   | Bytes               | **The difficulty of the PoW puzzle.**                        |
+| `hash`         | H256(hash)          | **The block hash.** Note that although this value is returned by the JSON-RPC API, it is in fact NOT a part of the header. This value is actually calculated from the `Header` data. |
+| `number`       | uint64              | **The block height.**                                        |
+| `parent_hash`  | H256(hash)          | **The hash of the parent block.**                            |
+| `seal`         | `nonce` and `proof` | **The seal of a block.** After finished the block assembling, the miner can start to do the calculation for finding the solution of the PoW puzzle. The "solution" here is called `seal`. |
+| `seal.nonce`   | uint64              | **The nonce.** Similar to [the nonce in Bitcoin](https://en.bitcoin.it/wiki/Nonce). |
+| `seal.proof`   | Bytes               | **The solution of the PoW puzzle.**                          |
```

Comment:
It's better to create a new table for `Seal` type.

Reply:
I don think it's necessary, same reason to the `input`.

It only increase the complexity.